### DISCUSSION TITLE
Meta property that shows if a queried page is the final page of the query

### DISCUSSION THREAD

**Discussion by shipsides-sparknel** [directus/directus#19070] [Feature Requests]
### Summary

When using the page global query parameter, it does not include any metadata to determine how many pages are left and/or if the page you're on is the last page.

You can work around this by using the meta property to determine the number of loops required and write a script accordingly, or keep hitting the API until you get data = [], but the best implementations of this have a meta property such as final_page = true

Past discussions that have relevance: #6609, #5348, #4247

### Motivation

If the standard cloud product continues to implement additionally restrictive limits to do with

- # of API queries
- max items read per request
- max returned size per request

API queries with pagination are going to be more important. Currently limit=-1 works perfectly for a lot of use cases **(refer to Alternatives section for further discussion)** and if this is taken away pagination is the alternative, and it feels wasteful and not very efficient to keep firing requests until the body is empty. Needing to script a does page number * item limit = meta.total_count and evaluate this before sending another request is less intuitive than a property that has already evaluated this.

### Detailed Design

New API query meta property for is_final_page

### Must Have:
- A way to determine if there is or isn't any additional data to fetch without re-hitting the API

### Could Have:
- Total of number of pages the query produces

### Drawbacks

Beyond committing resources to the implementation and taking them off higher priority work, it's tough to see any drawbacks.

### Alternatives

The alternative would be not (aggressively) restricting API read limits for cloud instances.

Allowing the use of limit = -1 as long as the query stays under the maximum allowed MB's would bypass the need for pagination for 99.99999% of API queries and make the importance of this feature requests significantly less.

I am not sure how a 50 object wide query that goes four relations deep and 5MB large is less resource intensive than reading a two object collection with 10000 items that's raw size is 20% less.

### Adoption Strategy

Any requests that have meta.* would have an extra property, whether or not that's a breaking change for those users depends on how they process the relevant properties.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
